sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -469,6 +470,40 @@ private[hive] class HiveMetastoreCatalog(val client: ClientInterface, hive: Hive
     }
   }
 
+  object WriteToDirs extends Rule[LogicalPlan] with HiveInspectors {
+    import org.apache.hadoop.hive.ql.Context
+    import org.apache.hadoop.hive.ql.parse.{ASTNode, SemanticAnalyzer}
+
+    def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+      // Wait until children are resolved.
+      case p: LogicalPlan if !p.childrenResolved => p
+
+      case WriteToDirectory(path, child, isLocal, extra: ASTNode) =>
```

Comment:
We shouldn't be holding onto the AST anymore, as this ties us a little too closely to a specific version of Hive.  Instead look at how we are doing things in [CreateTableAsSelect](https://github.com/apache/spark/blob/master/sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateTableAsSelect.scala).

Also, please add a test case where you use this new operation on a Spark SQL temporary table.